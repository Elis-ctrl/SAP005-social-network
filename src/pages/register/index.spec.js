import { Register } from './index.js';
import * as services from '../../services/index.js';

describe('Register', () => {
  beforeEach(() => {
    services.createProfile = jest.fn(() => Promise.resolve(true));
    services.confirmEmail = jest.fn(() => Promise.resolve(true));
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should be a function', () => {
    expect(typeof Register).toBe('function');
  });
  it('shoul return registerpag', () => {
    expect(Register()).toMatchSnapshot();
  });
  it('should click submmit button register', () => {
    const registerTest = Register();
    const btnRegister = registerTest.querySelector('#btnRegister');
    registerTest.querySelector('#email').value = 'email@teste1.com';
    registerTest.querySelector('#password').value = 'teste1';
    btnRegister.dispatchEvent(new Event('click'));
    expect(services.createProfile).toHaveBeenCalledWith('email@teste1.com', 'teste1');
  });
});
