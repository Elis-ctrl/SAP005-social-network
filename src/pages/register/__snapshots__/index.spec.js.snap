// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register shoul return registerpag 1`] = `
<div
  class="div-register"
>
  
  
  <div
    alt="Runner|Cadastro"
    class="head-register"
  >
    Runner|Cadastro
  </div>
  
  
  <figure
    class="container-logo"
  >
    
    
    <img
      alt="Logo Runners"
      class="logo-register"
      src="./assets/logo_runners.png"
    />
    
  
  </figure>
  

  
  <form>
     
    
    <div
      class="form-container"
    >
      
    
      <h3
        class="error"
        id="registerError"
      />
      
      
      <input
        id="displayName"
        placeholder="Nome"
        type="text"
      />
      
      
      <input
        id="email"
        placeholder="E-mail"
        type="email"
      />
      
      
      <input
        id="phoneNumber"
        placeholder="()999999999"
        type="number"
      />
      
      
      <input
        id="password"
        placeholder="Senha"
        type="password"
      />
      
      
      <input
        id="confirmPwd"
        placeholder="Confirmar Senha"
        type="password"
      />
      
      
      <button
        id="btnRegister"
      >
        Registrar
      </button>
      
    
      <a
        class="link-exit"
        href="/"
      >
        Voltar
      </a>
       
    
    </div>
    
  
  </form>
  
  
</div>
`;
